area_detector_handlers/handlers.py

```diff
@@ -20,6 +20,50 @@ class IntegrityError(Exception):
     pass
 
 
+class HDF5VariableFramesHandler(HandlerBase):
+    """
+    Handler for data stored in one Dataset of an HDF5 file.
+
+    Parameters
+    ----------
+    filename : string
+        path to HDF5 file
+    swmr : bool, optional
+        Open the hdf5 file in SWMR read mode. Only used when mode = 'r'.
+        Default is False.
+    """
+
+    specs = {"AD_HDF5_v1"} | HandlerBase.specs
+
+    def __init__(self, filename):
+        self._filename = filename
+        self._key = "/entry/data/data"
+        self._file = None
+        self._dataset = None
+        self.open()
+
+    def get_file_list(self, datum_kwarg_gen):
+        return [self._filename]
+
+    def __call__(self, offset, num_frames):
+        return self._dataset[offset : offset + num_frames]
+
+    def open(self):
```

Comment:
Make this method internal?

Reply:
I am not sure, maybe. This class was copied from https://github.com/bluesky/area-detector-handlers/blob/316e126663772f0a0597844c4184d4dc0800039e/area_detector_handlers/handlers.py#L84-L132

and then modified. I guess it never was a part of the public API. Then probably the `self.close()` should be updated similarly.

Reply:
No `close` is used by the base class in https://github.com/bluesky/area-detector-handlers/blob/316e126663772f0a0597844c4184d4dc0800039e/area_detector_handlers/__init__.py#L25-L29

and it part of the optional public API for handlers (see bottom of [this section](https://blueskyproject.io/event-model/external.html#handler-interface)).

I believe that separating the file-opening logic out from `__init__` into a separate method `open` was just intended to make it easier to override that specific logic in a subclass. There's little harm in leaving it public, really, since handlers are internal-ish objects anyway, but it would also be fine to make it private to be clear that it's not necessary/expected for user code to call it.

Reply:
Got it, I'll change it later.

Reply:
This is done via 47668be. I apologize for the delay.